packages/swingset-liveslots/src/watchedPromises.js

```diff
@@ -121,20 +145,28 @@ export function makeWatchedPromiseManager({
     }
   }
 
+  /**
+   * @template V
+   * @template {any[]} A]
+   * @param {import('./vatDataTypes.js').DurableKindHandle} kindHandle
+   * @param {(value: V, ...args: A) => void} fulfillHandler
+   * @param {(reason: any, ...args: A) => void} rejectHandler
+   * @returns {import('./types.js').PromiseWatcher<V, A>}
+   */
   function providePromiseWatcher(
     kindHandle,
-    fulfillHandler = x => x,
-    rejectHandler = x => {
-      throw x;
```

Comment:
@FUDCo removing this throw make sense to you? I think when when a rejectHandler throws that ends up as an unhandled promise rejection

Reply:
At first glance I thought you were right, but on further inspection I'm not sure that's so (though I'm not deeply confident of that conclusion).  In particular, it looks to me like there's a call chain from `startVat` that can result in somebody catching the exception.

On the other hand, I think the default `fulfillHandler` is questionable, since from the way it's invoked it doesn't look to me like there's ever going to be anybody positioned to receive the return value.